src/gui/guiutility.cpp

```diff
@@ -98,16 +99,36 @@ QString Utility::vfsFreeSpaceActionText()
     return QCoreApplication::translate("utility", "Free up local space");
 }
 
-QPixmap Utility::createColorAwareIcon(const QString &name, const QPalette &palette)
+QIcon Utility::createColorAwareIcon(const QString &name, const QPalette &palette)
 {
-    const QColor bg(palette.base().color());
-    QImage img(Theme::hidpiFileName(name));
-    if (img.isGrayscale()) {
+    const QIcon icon(name);
+    OC_ASSERT(!icon.isNull());
+    auto availableSize = icon.availableSizes();
+    if (availableSize.empty())
+    {
+        availableSize = {{48, 48}, {64, 64}, {128, 128}, {256, 256}};
+    }
+    QImage img = icon.pixmap(availableSize.first()).toImage();
+    if (!img.isGrayscale()) {
+        return icon;
+    } else {
```

Comment:
I guess we should just add a second set of colored svgs in to the pot and use them instead of this madness.